tests/ConfigFileTest.cpp

```diff
@@ -1014,16 +1014,28 @@ TEST_CASE_METHOD(TApp, "TOMLStringVector", "[config]") {
         std::ofstream out{tmptoml};
         out << "#this is a comment line\n";
         out << "[default]\n";
+        out << "zero1=[]\n";
+        out << "zero2={}\n";
+        out << "nzero={}\n";
+        out << "one=[\"1\"]\n";
         out << "two=[\"2\",\"3\"]\n";
         out << "three=[\"1\",\"2\",\"3\"]\n";
     }
 
-    std::vector<std::string> two, three;
+    std::vector<std::string> nzero, zero1, zero2, one, two, three;
+    app.add_option("--zero1", zero1)->required()->expected(0, 99)->default_str("{}");
+    app.add_option("--zero2", zero2)->required()->expected(0, 99)->default_val(std::vector<std::string>{});
```

Comment:
What happens if you add `->expected(0, 99)` but don't set default_str? Can whatever happens for that be tested?

Reply:
I added a test for that.  But that is basically the previous behavior and will result in a vector with a single empty string.

Reply:
The idea here is that there does need to be a way to get an empty vector as an output, but it shouldn't be default or easy.

Reply:
Seems like adding an empty string from a TOML `[]` is also problematic. Would is be possible to make this an error if a default is not specified? `[""]` would be a valid default to get the old behavior?